public/friendlycode/js/friendlycode.js

```diff
@@ -77,9 +81,36 @@ define(function(require) {
       editor.panes.preview.setViewLink(info.viewURL);
     });
 
+    // Bramble: Load the project Files into the fs
+    var initFs = function(callback) {
+      if(!makeDetails || !makeDetails.title) {
+        makeDetails = ProjectFiles.generateDefaultProject();
+        ProjectFiles.load(makeDetails, { defaultTemplate: defaultContent }, callback);
+        return;
+      }
+
+      if(makeDetails.isNew) {
+        makeDetails = ProjectFiles.generateDefaultProject(makeDetails.title);
+        ProjectFiles.load(makeDetails, {
+          isNew: true,
+          defaultTemplate: defaultContent,
+          csrfToken: $("meta[name='csrf-token']").attr("content"),
+          persistenceURL: options.appUrl + "/updateProjectFile"
+        }, callback);
+        return;
+      }
+
+      ProjectFiles.load(makeDetails, callback);
+    };
+
+    FileSystemSync.init(makeDetails && makeDetails.title, {
+      createOrUpdate: options.appUrl + "/updateProjectFile",
+      del: options.appUrl + "/deleteProjectFile"
+    }, $("meta[name='csrf-token']").attr("content"));
+
     if (!pageManager.currentPage()) {
       setTimeout(function() {
-        editor.panes.codeMirror.init(defaultContent);
+        editor.panes.codeMirror.init(defaultContent, initFs);
```

Comment:
I realize you're hacking this into the existing stuff, but for sure `editor.panes.codeMirror` is going away.  Think about how we'll do this with the new stuff.